### DISCUSSION TITLE
Сортировка треков

### DISCUSSION THREAD

**Discussion by alex69288** [Chimildic/YaMuTools#64] [Помощь с ошибкой] [answered]
Небольшая проблемка... Когда:

1. Заходишь на страницу исполнителя
2. Нажимаешь Дискография - Все треки
![image](https://user-images.githubusercontent.com/81667055/190219750-e53a32d1-dae7-4703-9b7b-1b436bec2617.png)
3. Создается плейлист с дискографией в обратном порядке (т.е. от новых треков к старым)
4. Нажимаю сортировка в обратном порядке
![image](https://user-images.githubusercontent.com/81667055/190221249-09b4c393-87cd-41d7-892c-6d705b05f038.png)

И далее, он сортирует только по альбомам, но сами треки в плейлисте, они так и остаются отсортированными от новых к старым
![image](https://user-images.githubusercontent.com/81667055/190221652-d8265aab-0f1e-422f-9695-3573dded4eb0.png)

И тут видно, что сами треки начинаются снизу.
![image](https://user-images.githubusercontent.com/81667055/190221735-e272ab20-55ef-4e0f-a314-560416887261.png)

Мелочь, конечно, для такого крутого приложения. Но все же, если получиться исправить, будет прямо идеал.

Еще кстати моментик,
Почему то если открываешь эти плейлисты на телефоне, там вообще все вперемешку, в то же время, этот же плейлист открытый через браузерную версию, отсортирован нормально. Странно.

**Comment by F000x** [comment]
Я не разработчик этого расширения, но могу сказать следующее. У Я.Музыки на страницах исполнителей треки отсортированы не по хронологии, а по популярности, т.к. треки из одного альбома могут быть и в начале и в конце страницы. В вашем примере https://music.yandex.ru/artist/4990388/tracks треки из альбома Энтузиаст находяся на 1 и на 46 месте, т.е. хронологии от старых к новым или наоборот тут нет. Расширение же, насколько я понимаю, собирает дискографию по альбомам. По-моему мнению, это удобнее, т.к. легче найти нужный трек в дискографии, и получается как раз-таки хронологический порядок - треки, выпущенные одновременно в одном альбоме, в дискографии находятся рядом.

То, что вы хотите сделать, можно сделать стандартными средствами Я.Музыки. Инструкцию для этого написал kisoid вот тут:  https://github.com/Chimildic/YaMuTools/discussions/54#discussioncomment-3332898 Тогда вы получите дискографию исполнителя в том порядке, в котором она находится на странице исполнителя.

По поводу отличий плейлистов в приложении и в веб-версии: синхронизация происходит не сразу, сам сталкивался с тем, что порядок треков и состав плейлиста не синхронизировались с веб-версией несколько часов. Чтобы ускорить процесс можно выйти из приложении и авторизоваться заново или очистить все данные в приложении и авторизоваться заново. Для себя я нашёл следующую схему: создаю плейлист в веб-версии с другого аккаунта, не с того, в котором авторизован в приложении. Беру ссылку на этот плейлист и перехожу по ней на телефоне, нажимая открыть в приложении. Тогда плейлист сразу соответствует созданному в веб-версии. А там его уже можно добавить в любимые или скопировать в самом приложении.

**Reply by alex69288** [reply]
> Я не разработчик этого расширения, но могу сказать следующее. У Я.Музыки на страницах исполнителей треки отсортированы не по хронологии, а по популярности, т.к. треки из одного альбома могут быть и в начале и в конце страницы. В вашем примере https://music.yandex.ru/artist/4990388/tracks треки из альбома Энтузиаст находяся на 1 и на 46 месте, т.е. хронологии от старых к новым или наоборот тут нет. Расширение же, насколько я понимаю, собирает дискографию по альбомам. По-моему мнению, это удобнее, т.к. легче найти нужный трек в дискографии, и получается как раз-таки хронологический порядок - треки, выпущенные одновременно в одном альбоме, в дискографии находятся рядом.
> 
> То, что вы хотите сделать, можно сделать стандартными средствами Я.Музыки. Инструкцию для этого написал kisoid вот тут: [#54 (comment)](https://github.com/Chimildic/YaMuTools/discussions/54#discussioncomment-3332898) Тогда вы получите дискографию исполнителя в том порядке, в котором она находится на странице исполнителя.
> 
> По поводу отличий плейлистов в приложении и в веб-версии: синхронизация происходит не сразу, сам сталкивался с тем, что порядок треков и состав плейлиста не синхронизировались с веб-версией несколько часов. Чтобы ускорить процесс можно выйти из приложении и авторизоваться заново или очистить все данные в приложении и авторизоваться заново. Для себя я нашёл следующую схему: создаю плейлист в веб-версии с другого аккаунта, не с того, в котором авторизован в приложении. Беру ссылку на этот плейлист и перехожу по ней на телефоне, нажимая открыть в приложении. Тогда плейлист сразу соответствует созданному в веб-версии. А там его уже можно добавить в любимые или скопировать в самом приложении.

На счет сортировки, мне кажется, что расширение в как раз таки собирает дискографию по альбомом, а не по списку треков из вкладки треки, потому что сами альбомы в плейлисте отсортированы по хронологии.
У Яндекс.Музыки треки по популярности, отсортированы только на в вкладке треки, а в самих альбомах, треки в хронологическом порядке (на сколько мне известно).

За инструкцию спасибо, но в как раз то что я хочу сделать, с помощью этой инструкции, сделать не получиться, потому что отсортировать в хронологическом порядке таким способом не выйдет. Этим способом получиться создать простой плейлист с треками отсортированными по популярности, ну или же с другой сортировкой предоставленной Яндексом.

На счет приложения, странно, потому что порядок треков там все равно такой же, но вот обложки сами меняются, я так понял что это происходит из-за того, что приложение недавно обновили (или давно), и дискография теперь выглядит немного по другому, т.е. есть сами альбомы, и отдельно сама дискография, где находятся треки со сборников видимо, как то так, мне кажется.
Еще в приложении есть такой момент, в созданном плейлисте, есть одинаковое треки, например трек из альбома, и трек из какого-нибудь сборника, и когда ты воспроизводишь какой-нибудь из них двоих, то по крайней мере в приложении, они оба отображены как проигрывающиеся, т.е. у тебя в одно и то же время проигрывается два трека. Это никак не мешает, это уже ошибки Яндекса, ну или фичи.

А так в целом, приложение очень крутое. Возможности прослушать отсортированную дискографию, прямо не хватало.

**Comment by Chimildic** [comment, marked as answer]
Сортировка в обратном порядке не смотрит на альбомы в отдельности. Поэтому все переворачивается. 

В новой версии расширения добавил выбор сортировки непосредственно перед сбором дискографии. Ожидайте обновления.

Отмечу сразу, что даты добавления в лайки неизвестны. Поэтому с такой сортировкой ничего не сделать.

**Comment by alex69288** [comment]
Спасибо огромное, за выполненое пожелание) Теперь с сортировкой все круто)
Мне сильно не хватало такой возможности, спасибо)

**Comment by sergvpr** [comment]
Можно ли реализовать вариант сортировки треков в плейлисте исходя из истории прослушивания?
Я знаю, что есть инструмент "удалить недавно игравшие", но для этого нужно создавать отдельный плейлист и там удалять историю, чтобы не терять треки в текущем плейлисте. 
Поэтому и возникла эта идея сортировать текущий плейлист - брать треки из истории прослушивания и сортировать от менее прослушиваемых к более.
удобство, как говорится, на лицо.
